lfucg-exactions/client/js/components/DeclineDelete.js

```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import {
+    hashHistory,
+} from 'react-router';
+
+import {
+    postDelete,
+} from '../actions/apiActions';
+
+class DeclineDelete extends React.Component {
+    componentDidMount() {
+        this.props.onComponentDidMount({
+            currentForm: this.props.currentForm,
+            selectedEntry: this.props.selectedEntry,
+            parentRoute: this.props.parentRoute,
+        });
+    }
+
+    render() {
+        const {
+            currentUser,
+            onDelete,
+        } = this.props;
+
+        return (
+            <div className="decline-delete">
+                {currentUser && (currentUser.is_superuser || (currentUser.profile && currentUser.profile.is_supervisor)) &&
+                    <button className="btn btn-danger btn-lex-danger" data-toggle="modal" data-target="#deleteConfirm">
+                        Decline / Delete
+                    </button>
+                }
+                <div id="deleteConfirm" className="modal fade" role="dialog">
+                    <div className="modal-dialog">
```

Comment:
does react not have any modal dialog components that can be used instead of rolling your own?